pandas/core/reshape/merge.py

```diff
@@ -1679,11 +1679,6 @@ def _convert_to_multiindex(index: Index) -> MultiIndex:
     # lindexer and rindexer hold the indexes where the join occurred
     # for left and right respectively. If left/right is None then
     # the join occurred on all indices of left/right
-    if lindexer is None:
```

Comment:
Is the above comment relevant/can be removed?

Reply:
I've removed the comment now that we're not creating a range of all indices.